lib/src/fnc/type.rs

```diff
@@ -46,10 +49,15 @@ pub fn int((arg,): (Value,)) -> Result<Value, Error> {
 }
 
 pub fn number((arg,): (Value,)) -> Result<Value, Error> {
-	match arg {
-		Value::Number(_) => Ok(arg),
-		_ => Ok(Value::Number(arg.as_number())),
-	}
+	Ok(match &arg {
+		#[cfg(debug_assertions)]
+		Value::Number(Number::Float(f)) => {
+			assert!(!f.is_nan(), "unexpected NAN float in fnc::type::number");
```

Comment:
@finnbear out of interest why `assert` here, but `debug_assert` higher up?

Reply:
As this is within a block annotated by `#[cfg(debug_asssertions)]`, it is a `debug_assert!` in disguise. The motivation is to possibly reduce conditional branches in release mode.

Reply:
Let me know if you think `debug_assert` (just as efficient as the redundant check is at compile time) would be better/clearer.

Reply:
Ah yes didn't spot that. I just saw it as 'different' from the one above... Should we not make them work the same wa

Reply:
Fair enough! When I get home, I'll refactor things to increase consistency (hopefully) without compromising performance.